### PR TITLE
feat: add a check to ensure logstore is enabled

### PR DESCRIPTION
**Pull request by keevan**
image

image

### CODE REVIEW

File: classes/check/logstorecheck.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+/**
+ * Range request performance check.
+ *
+ * @package   tool_heartbeat
+ * @author    Kevin Pham <kevinpham@catalyst-au.net>
+ * @copyright Catalyst IT, 2022
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ *
+ * This can be run either as a web api, or on the CLI. When run on the
```
Comment (brendanheywood) [inline classes/check/logstorecheck.php:29]: this fluff can probably not be here but no worries either way

Comment (keevan) [reply classes/check/logstorecheck.php]: fluff -> gone!

File: classes/check/logstorecheck.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    /**
+     * Process and return whether there is one or more logstores
+     *
+     * @return result of the check
+     */
+    public function get_result(): result {
+        // Define possible responses.
+        $ok = new result(result::OK, get_string('checklogstoreok', 'tool_heartbeat'));
+        $bad = new result(result::ERROR, get_string('checklogstorebad', 'tool_heartbeat'));
```
Comment (brendanheywood) [inline classes/check/logstorecheck.php:49]: I'd make this result::CRITICAL over ERROR

Comment (keevan) [reply classes/check/logstorecheck.php]: Converted result to CRITICAL

File: classes/check/logstorecheck.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tool_heartbeat\check;
+use core\check\check;
+use core\check\result;
+
+/**
+ * Range request performance check.
```
Comment (brendanheywood) [inline classes/check/logstorecheck.php:22]: cut and paste

Comment (keevan) [reply classes/check/logstorecheck.php]: Good catch. Resolved

**Pull request merged**